src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/WindowsCounters.cs

```diff
@@ -10,203 +11,310 @@ internal sealed class WindowsCounters
 {
     public WindowsCounters(IMeterFactory meterFactory, TcpTableInfo tcpTableInfo)
     {
+        const string NetworkStateKey = "system.network.state";
+        const string GaugeName = "process.network.connections";
 #pragma warning disable CA2000 // Dispose objects before losing scope
         // We don't dispose the meter because IMeterFactory handles that
         // An issue on analyzer side: https://github.com/dotnet/roslyn-analyzers/issues/6912
         // Related documentation: https://github.com/dotnet/docs/pull/37170
         var meter = meterFactory.Create("Microsoft.Extensions.Diagnostics.ResourceMonitoring");
 #pragma warning restore CA2000 // Dispose objects before losing scope
 
+        var tcpTag = new KeyValuePair<string, object?>("network.transport", "tcp");
+
+        // These are covered in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-metrics.md#attributes:
+        var tcpVersionFourTag = new KeyValuePair<string, object?>("network.type", "ipv4");
+        var tcpVersionSixTag = new KeyValuePair<string, object?>("network.type", "ipv6");
+
+        // TODO: these are heavily rewritten to align with
+        // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/system-metrics.md#metric-systemnetworkconnections
+        // All these should be of UpDownCounter type
```

Comment:
@noahfalk @reyang I'm mostly worried about these gauges. Should I make them `UpDownCounter` instead or use `process.network.connections.count` name (to continue keeping them as gauges). What do you think?

Reply:
+1 These should be `UpDownCounter`s. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#instrument-selection

The current name (`process.network.connections`) seems _better_ than `process.network.connections.count` in my personal opinion - although OTel semantic convention has the final saying. (this is another example: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#metric-httpserveractive_requests, note that I didn't use a permanent link, since the main branch version is very likely to be the final stable version)

Reply:
Changed to `UpDownCounter`s, thanks